File: reactapp4.client/src/interfaces/SportRadarMLBPBPAtBat.ts
export interface SportRadarMLBPBPAtBat {
  game_id: string | null;                // uuid
  inning: number | null;
  half: string | null;                   // "T"/"B"
  hitter_id: string | null;              // uuid
  at_bat_id: string | null;              // uuid
  hitter_hand: string | null;            // "L"/"R"/"S"
  pitcher_id: string | null;             // uuid
  pitcher_hand: string | null;           // "L"/"R"/"S"
  sequence_number: number | null;
  description: string | null;

  hitter_preferred_name: string | null;
  hitter_first_name: string | null;
  hitter_last_name: string | null;
  hitter_jersey_number: number | null;
  hitter_full_name: string | null;

  pitcher_preferred_name: string | null;
  pitcher_first_name: string | null;
  pitcher_last_name: string | null;
  pitcher_jersey_number: number | null;
  pitcher_full_name: string | null;

  home_team_runs: number | null;
  away_team_runs: number | null;
}

File: ReactApp4.Server/Data/SportRadarMLBPBPAtBat.cs
using System;

namespace ReactApp4.Server.Data
{
    public class SportRadarMLBPBPAtBat
    {
        public Guid? GameId { get; set; }
        public int? Inning { get; set; }
        public string? Half { get; set; }                    // "T" or "B"
        public Guid? HitterId { get; set; }
        public Guid? AtBatId { get; set; }
        public string? HitterHand { get; set; }              // "L","R","S"
        public Guid? PitcherId { get; set; }
        public string? PitcherHand { get; set; }             // "L","R","S"
        public int? SequenceNumber { get; set; }
        public string? Description { get; set; }

        public string? HitterPreferredName { get; set; }
        public string? HitterFirstName { get; set; }
        public string? HitterLastName { get; set; }
        public int? HitterJerseyNumber { get; set; }
        public string? HitterFullName { get; set; }

        public string? PitcherPreferredName { get; set; }
        public string? PitcherFirstName { get; set; }
        public string? PitcherLastName { get; set; }
        public int? PitcherJerseyNumber { get; set; }
        public string? PitcherFullName { get; set; }

        public int? HomeTeamRuns { get; set; }
        public int? AwayTeamRuns { get; set; }
    }
}
